Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -76,49 +85,64 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
-                }
+        guard let source = source, bookmark = source.current[indexPath.row] else { return super.tableView(tableView, cellForRowAtIndexPath: indexPath) }
+        let cell: UITableViewCell
+        if let _ = bookmark as? BookmarkFolder {
+            cell = tableView.dequeueReusableCellWithIdentifier(BookmarkFolderCellIdentifier, forIndexPath: indexPath)
+        } else {
+            cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
+            if let url = bookmark.favicon?.url.asURL where url.scheme == "asset" {
+                cell.imageView?.image = UIImage(named: url.host!)
+            } else {
+                cell.imageView?.setIcon(bookmark.favicon, withPlaceholder: self.defaultIcon)
+            }
+        }
 
-                switch (bookmark) {
-                    case let item as BookmarkItem:
-                        if item.title.isEmpty {
-                            cell.textLabel?.text = item.url
-                        } else {
-                            cell.textLabel?.text = item.title
-                        }
-                    default:
-                        // Bookmark folders don't have a good fallback if there's no title. :(
-                        cell.textLabel?.text = bookmark.title
+        switch (bookmark) {
+            case let item as BookmarkItem:
+                if item.title.isEmpty {
+                    cell.textLabel?.text = item.url
+                } else {
+                    cell.textLabel?.text = item.title
                 }
-            }
+            default:
+                // Bookmark folders don't have a good fallback if there's no title. :(
+                cell.textLabel?.text = bookmark.title
         }
 
         return cell
     }
 
-    func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
         // Don't show a header for the root
-        if source == nil || source?.current.guid == BookmarkRoots.MobileFolderGUID {
+        if source == nil || source?.current.guid == BookmarkRoots.RootGUID {
             return nil
         }
+        guard let header = tableView.dequeueReusableHeaderFooterViewWithIdentifier(BookmarkFolderHeaderViewIdentifier) as? BookmarkFolderTableViewHeader else { return nil }
 
-        // Note: If there's no root (i.e. source == nil), we'll also show no header.
-        return source?.current.title
+        // register as delegate to ensure we get notified when the user interacts with this header
+        if header.delegate == nil {
+            header.delegate = self
+        }
+
+        if let parentFolder = parentFolder {
+            if parentFolder.guid == BookmarkRoots.RootGUID {
+                header.textLabel?.text = NSLocalizedString("Bookmarks", comment: "Panel accessibility label")
```

Comment:
We're past string freeze for 1.1, right? Did this string already land somewhere else?